Source/SecureChat/SecureChat.swift

```diff
@@ -0,0 +1,474 @@
+
+        return CallbackOperation { _, completion in
+            do {
+                Log.debug("Starting new session with \(receiverCard.identity)")
+
+                guard self.existingSession(withParticpantIdentity: receiverCard.identity) == nil else {
+                    throw SecureChatError.sessionAlreadyExists
+                }
+
+                let tokenContext = TokenContext(service: "ratchet", operation: "get")
+                let getTokenOperation = OperationUtils.makeGetTokenOperation(
+                    tokenContext: tokenContext, accessTokenProvider: self.accessTokenProvider)
+
+                let token = try getTokenOperation.startSync().getResult()
+
+                // TODO: Check long-term and one-time public keys are X25519
+                let publicKeySet = try self.client.getPublicKeySet(forRecipientIdentity: receiverCard.identity,
+                                                                   token: token.stringRepresentation())
+
+                // TODO: Check identityPublicKey is ED25519
+                guard let identityPublicKey = receiverCard.publicKey as? VirgilPublicKey else {
+                    throw SecureChatError.wrongIdentityPublicKeyCrypto
+                }
+
+                guard self.crypto.exportPublicKey(identityPublicKey) == publicKeySet.identityPublicKey else {
+                    throw SecureChatError.identityKeyDoesntMatch
+                }
+
+                guard self.crypto.verifySignature(publicKeySet.longTermPublicKey.signature,
+                                                  of: publicKeySet.longTermPublicKey.publicKey,
+                                                  with: identityPublicKey) else {
+                    throw SecureChatError.invalidLongTermKeySignature
+                }
+
+                if publicKeySet.oneTimePublicKey == nil {
+                    Log.error("Creating weak session with \(receiverCard.identity)")
+                }
+
+                let privateKeyData = self.crypto.exportPrivateKey(self.identityPrivateKey)
+
+                let session = try SecureSession(sessionStorage: self.sessionStorage,
+                                                participantIdentity: receiverCard.identity,
+                                                senderIdentityPrivateKey: privateKeyData,
+                                                receiverIdentityPublicKey: publicKeySet.identityPublicKey,
+                                                receiverLongTermPublicKey: publicKeySet.longTermPublicKey.publicKey,
+                                                receiverOneTimePublicKey: publicKeySet.oneTimePublicKey)
+
+                try self.sessionStorage.storeSession(session)
+
+                completion(session, nil)
+            }
+            catch {
+                completion(nil, error)
+            }
+        }
+    }
+
+    private let queue = DispatchQueue(label: "VSRSecureChat", qos: .background)
```

Comment:
Variable declaration in the middle of class. Should be before methods.

Reply:
Accordingly to what?